### ISSUE TITLE
Lint/ShadowingOuterLocalVariable false positive in presence of numbered parameters

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#11444]
This code:
```ruby
def foo(data)
  data.map {
    current, prev = *_1
    [current, prev - current]
  }.select { |current, change|
    change > 10
  }
end
```
...produces false `Lint/ShadowingOuterLocalVariable` offense.

## Expected behavior

No `Lint/ShadowingOuterLocalVariable` reported

## Actual behavior

```
$ rubocop rbcp.rb -d
For /home/zverok/playground/ruby: configuration from /home/zverok/playground/ruby/.rubocop.yml
Default configuration from /home/zverok/.rvm/gems/ruby-head/gems/rubocop-1.43.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/zverok/playground/ruby/rbcp.rb
Loading cache from /home/zverok/.cache/rubocop_cache/923f4888f7586f2e6600a5d5e5f0b67a54e01c76/6d7a3b621ca1730e04accd938619e4bdab66cfb1/dac28bdd688df14856e4ef5fec73379bc1153730
W

Offenses:

[...skipping unrelated...]

rbcp.rb:5:15: W: Lint/ShadowingOuterLocalVariable: Shadowing outer local variable - current.
  }.select { |current, change|
              ^^^^^^^
```

## Steps to reproduce the problem

The only thing that needs to be in `.rubocop.yml` to reproduce is 
```yaml
AllCops:
  TargetRubyVersion: 3.2 # or any other starting from 2.7
```
...because the bug manifests in presence of numbered parameters only. This would not produce the offense:
```ruby
def foo(data)
  data.map { |v|
    current, prev = *v
    [current, prev - current]
  }.select { |current, change|
    change > 10
  }
end
```

## RuboCop version
```
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [x86_64-linux]
```
